src/include/rbd/librbd.h

```diff
@@ -385,6 +385,13 @@ typedef enum {
 
 typedef void *rbd_encryption_options_t;
 
+typedef struct {
+    bool is_encrypted;
+    rbd_encryption_format_t format;
+    rbd_encryption_options_t opts;
+    size_t opts_size;
+} rbd_encryption_spec_t;
```

Comment:
It is currently used by the CLI (by `struct EncryptionOptions` in tools/rbd/Utils).
It will later be used by new APIs such as `migration_execute_with_encryption`, etc

Reply:
I am not very happy with adding new structures to API, when it is not necessary.

The problem with a structure is that you can't extend it later if you need. You will have to introduce a new structure and all new functions to use it. It seems to me `migration_execute_with_encryption` could just use `rbd_encryption_format_t format,  rbd_encryption_options_t opts,  size_t opts_size` arguments.

Just IMHO, I might change my mind when I saw real examples of usage of `rbd_encryption_spec_t` by API functions.

Reply:
It is not likely that we'll want to extend `rbd_encryption_spec_t`, since we have `rbd_encryption_options_t opts, size_t opts_size` inside, which were built to be extendable.

Without this struct, the signature of API functions taking source+dest encryptions arguments will be much more cumbersome:

`migration_execute_with_encryption(..., encryption_format_t src_encryption_format, encryption_options_t src_encryption_options, size_t src_encryption_options_size, encryption_format_t dst_encryption_format, encryption_options_t dst_encryption_options, size_t dst_encryption_options_size)`

instead of:

`migration_execute_with_encryption(..., encryption_spec_t src_encryption,  encryption_spec_t dst_encryption)`

and perhaps even further arguments will have to be added to support:
1. plaintext source or dest
2. copy src encryption as-is (crypto header) to dest

These are already supported by `rbd_encryption_spec_t`.

Reply:
ack